src/librustdoc/clean/mod.rs

```diff
@@ -1480,8 +1480,88 @@ pub(crate) fn clean_middle_assoc_item<'tcx>(
     Item::from_def_id_and_parts(assoc_item.def_id, Some(assoc_item.name), kind, cx)
 }
 
+/// The goal of this function is to return the first `Path` which is not private (ie not private
+/// or `doc(hidden)`). If it's not possible, it'll return the "end type".
+///
+/// If the path is not a re-export or is public, it'll return `None`.
+fn first_not_private(
```

Comment:
Minor, sounds better (I'm not a native speaker tho).

```suggestion
fn first_non_private(
```

Reply:
Sounds better indeed, let's go with it!